testng-core/src/test/java/test/thread/ParallelSuiteTest.java

```diff
@@ -79,10 +59,10 @@ public void suitesShouldRunInParallel4() {
         5,
```

Comment:
This test implies excluding duplicating suite files from run, which current code DOES NOT. Should we really exclude them? Or just change test case expected results. 
My opinion  would be to keep this as it is, since no one complained since 2011. @krmahadevan @juherr 

More details:
`expectedThreadCount` and `expectedSuiteCount` is 5 in this test.

However suite graph is created with 8 nodes, resulting in 8 suites(with duplicates) run, as follows:

```
suite-parallel-1.xml suite-parallel-2.xml suite-parallel-2-1.xml suite-parallel-2-2.xml
                                |                                            |
                                |                                            |
                      ---------- ---------|                                  v
                      v                   v                         suite-parallel-2-2-1.xml
            suite-parallel-2-1.xml suite-parallel-2-2.xml
                                                     |
                                                     |
                                                     v
                                            suite-parallel-2-2-1.xml
```

Reply:
@dsankouski - I dont think anyone in the real world would be trying to run the same suite multiple times by passing it multiple times. That perhaps is why no one complained. But assuming if anyone has been doing it and TestNG never complained, pruning for duplicates is going to cause complaints. So as you suggest, I also agree that we should just leave it as is (dont prune the duplicate suites), but just adjust the expectations of the test to satisfy the current behaviour.

@juherr WDYT ?

Reply:
I loved that visual representation :) Helps clarify the question a lot @dsankouski

Reply:
@juherr , pushing changed expected results solution, while waiting for you reply to check if it works and CI passes.

Reply:
I don't understand why the current expectedSuiteCount is 5.
If not 8, it should be 4, right?

Reply:
distinct suite name count is 5 in the graph: `suite-parallel-1.xml suite-parallel-2.xml suite-parallel-2-1.xml suite-parallel-2-2.xml suite-parallel-2-2-1.xml`

Reply:
Ok, got it. Thanks for the clarification.

8 or 5 looks good for me. 

@krmahadevan you choose.